bytecomp/matching.ml

```diff
@@ -1295,6 +1344,47 @@ and dont_precompile_var args cls def k =
     k )
 
 and precompile_or argo cls ors args def k =
+  (* Let's say the following pm was given to [split_and_precompile]
+     {[
+       match e1           ,  e2 ,  e3 with
+       | C1(arg11, arg21) ,  p1 ,  q1 -> lam1
+       | C2(arg12)        ,  p2 ,  q2 -> lam2
+       | ( C1(_, argO)
+         | C2(argO)   )   ,  p3 ,  q3 -> lam3
+       | C3               ,  p4 ,  q4 -> lam4
+       | C1(_, _)         ,  _  ,  _  -> lam5
+     ]}
+
+     Assuming the constructors are not for an extensible type, then
+     after [split_or], we end up here with:
+     {[
+       cls = [ ((C1, [ arg11; arg21 ; p1 ; q1]), lam1)
+             ; ((C2, [ arg12; p2; q2 ]), lam2)
+             ; ((C3, [ p4; q4 ]), lam4)
+             ]
```

Comment:
I don't remember, are patterns really split into a head and the arguments (pushed into the row) at this point?

Reply:
They are in `cls` yes.